src/__experimental__/components/date/date.component.js

```diff
@@ -116,9 +100,13 @@ class BaseDateInput extends React.Component {
 
   reformatVisibleDate = () => {
     const { visibleValue } = this.state;
-
     if (DateHelper.isValidDate(visibleValue)) {
-      this.setState({ visibleValue: formatDateToCurrentLocale(visibleValue) });
+      this.setState({ visibleValue: formatDateToCurrentLocale(visibleValue) }, () => {
+        if (this.props.onBlur) {
+          const dateWithSlashes = DateHelper.sanitizeDateInput(visibleValue);
+          this.props.onBlur({ target: this.input }, DateHelper.formatValue(dateWithSlashes));
```

Comment:
I see that this `this.props.onBlur()` call was originally in the `handleBlur()` method, but has been moved here.

The call is now conditional on `if (DateHelper.isValidDate(visibleValue)) {` so if that condition is `false` then `this.props.onBlur()` won't be called when a `blur` event occurs. Is that the intended behaviour?

Reply:
You're right I don't think the onBlur should be conditional on this, will refactor

Reply:
Thanks for moving the `this.props.onBlur()` call back into the `handleBlur()` method :thumbsup: